pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1664,6 +1664,9 @@ func (r *RawArtifact) HasLocation() bool {
 type HTTPArtifact struct {
 	// URL of the artifact
 	URL string `json:"url" protobuf:"bytes,1,opt,name=url"`
+
+	// Optional headers to include with the HTTP request for the artifact.
+	Headers map[string]string `json:"headers,omitempty" protobuf:"bytes,2,opt,name=headers"`
```

Comment:
I think we need to have a "map vs list" discussion to have here. Kubersetens YAMLs typically have lists rather than maps and use strategic-merge-patch with the `name` to merge them together. But with HTTP headers, it is allowed (but now common) to repeat headers - but I propose we do NOT support duplicate headers. 

@jessesuen you'll need to approve this change - thoughts?

Can you run `make codegen` to fix the build?

Reply:
> Can you run `make codegen` to fix the build?

Done 😄 

> But with HTTP headers, it is allowed (but now common) to repeat headers - but I propose we do NOT support duplicate headers.

I assume this means we'd use the `map` type like I have now, since obviously a map cannot duplicate keys...? Am I interpreting your comment correctly?

Reply:
kind of -  I want @jessesuen to make the call

Reply:
Spoken to @jessesuen today. We may want to be able to set headers from secrets in the future, and map would not allow this. Instead, can you change this to 

```yaml
- name: Authorization
  value: Bearer ****
```

In the future this could be:

```yaml
- name: Authorization
  valueFrom:
    secretKeyRef:
       name: my-secrets
       value: auth-token
```

Reply:
@alexec OK, makes sense - updated the PR such that an `HTTPArtifact` can be initialized with an optional list of headers a la:

```golang
// Header indicate a key-value request header to be used when fetching artifacts over HTTP
type Header struct {
	// Name is the header name
	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`

	// Value is the literal value to use for the header
	Value string `json:"value" protobuf:"bytes,2,opt,name=value"`
}

// HTTPArtifact allows an file served on HTTP to be placed as an input artifact in a container
type HTTPArtifact struct {
	// URL of the artifact
	URL string `json:"url" protobuf:"bytes,1,opt,name=url"`

	// Headers are an optional list of headers to send with HTTP requests for artifacts
	// +patchStrategy=merge
	// +patchMergeKey=name
	Headers []Header `json:"headers,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,2,opt,name=headers"`
}
```